tests/lib/rules/no-implicit-injections.js

```diff
@@ -0,0 +1,369 @@
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-implicit-injections');
+const RuleTester = require('eslint').RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: require.resolve('@babel/eslint-parser'),
+  parserOptions: {
+    ecmaVersion: 2020,
+    sourceType: 'module',
+    babelOptions: {
+      configFile: require.resolve('../../../.babelrc'),
+    },
+  },
+});
+
+ruleTester.run('no-implicit-injections', rule, {
```

Comment:
Since we're using the class stack, technically we should also be testing that the behavior is correct with nested classes. For example, that the behavior is correct inside the nested class, and after the nested class but still inside the outer class.

Reply:
I'll look at other test scenarios for class stack to see how those setup tests.

Reply:
This actually lead to finding some failures.

`ClassDeclaration` alone doesn't properly identify nested classes nor modules returned by functions.

This lead to a few more changes because of underlying issues in `ember/utils` commented in diff.